test/test_site.rb

```diff
@@ -494,9 +494,9 @@ def convert(*_args)
                                             source_dir, "_data", "categories.01", "dairy.yaml"
                                           ))
 
-        assert_equal site.data["categories01"]["dairy"], file_content
```

Comment:
Note: I couldnt figure out exactly where to apply a fix that would not cause risks to how collections currently work and figured that hopefully someone on the Jekyll team could help figure out a solution, otherwise this could be a breaking change to some users. I think this would be better done for data directory only instead of causing issues on all collections, which may mean multiple directories for each site a user has.